domain/src/main/java/org/oppia/domain/audio/AudioPlayerController.kt

```diff
@@ -65,28 +66,31 @@ class AudioPlayerController @Inject constructor(
    */
   class PlayProgress(val type: PlayStatus, val position: Int, val duration: Int)
 
-  private val mediaPlayer: MediaPlayer by lazy { MediaPlayer() }
+  private var mediaPlayer: MediaPlayer = MediaPlayer()
   private var playProgress: AudioMutableLiveData? = null
   private var nextUpdateJob: Job? = null
   private val audioLock = ReentrantLock()
 
   private var prepared = false
   private var observerActive = false
   private var mediaPlayerActive = false
+  private var isReleased = false
 
   private val SEEKBAR_UPDATE_FREQUENCY = TimeUnit.SECONDS.toMillis(1)
 
   /**
    * Loads audio source from a URL and return LiveData to send updates.
    * This controller cannot already be initialized.
    */
-  fun initializeMediaPlayer(url: String): LiveData<AsyncResult<PlayProgress>> {
+  fun initializeMediaPlayer(): LiveData<AsyncResult<PlayProgress>> {
     audioLock.withLock {
       check(!mediaPlayerActive) { "Media player has already been initialized" }
-      mediaPlayer.reset()
       mediaPlayerActive = true
+      if (isReleased) {
+        mediaPlayer = MediaPlayer()
```

Comment:
Why is recreation necessary?

Reply:
This recreation is necessary because you can't reset the mediaplayer after being released.